daemon/src/GHCSpecter/Driver.hs

```diff
@@ -0,0 +1,79 @@
+module GHCSpecter.Driver
+  ( driver,
+  )
+where
+
+import Control.Concurrent (forkIO, threadDelay)
+import Control.Concurrent.STM
+  ( TChan,
+    TVar,
+    atomically,
+    readTChan,
+    readTVar,
+    retry,
+    writeTChan,
+  )
+import Control.Lens ((^.))
+import Control.Monad.Extra (loopM)
+import Control.Monad.Trans.Reader (ReaderT (runReaderT))
+import Data.Time.Clock (nominalDiffTimeToSeconds)
+import GHCSpecter.Control qualified as Control (main)
+import GHCSpecter.Control.Runner (stepControlUpToEvent)
+import GHCSpecter.Server.Types
+  ( HasServerState (..),
+    ServerState (..),
+  )
+import GHCSpecter.UI.Constants (chanUpdateInterval)
+import GHCSpecter.UI.Types (UIState)
+import GHCSpecter.UI.Types.Event
+  ( BackgroundEvent (MessageChanUpdated),
+    Event,
+  )
+
+driver ::
+  (TVar UIState, TVar ServerState) ->
```

Comment:
These mutable variables should be tupled up as a separate data type with appropriate comments.

Reply:
ClientSession introduced.